ctms/crud.py

```diff
@@ -173,7 +176,22 @@ def get_contact_by_email_id(db: Session, email_id: UUID4) -> Optional[ContactSch
     email = get_email(db, email_id)
     if email is None:
         return None
-    return ContactSchema.from_email(email)
+    contact = ContactSchema.from_email(email)
+
+    # In the `ContactSchema` instance, the newsletters and waitlists are
+    # instances of `NewsletterSchema` and `WaitlistSchema`,
+    # which loose the `create_timestamp` and `update_timestamp` fields.
+    # Note: the whole code base with regards to models and schemas is a real house of cards.
+    contact_newsletters = get_newsletters_by_email_id(db, email_id)
+    contact.newsletters = [
+        NewsletterTableSchema.from_newsletter(nl) for nl in contact_newsletters
+    ]
+    contact_waitlists = get_waitlists_by_email_id(db, email_id)
+    contact.waitlists = [
+        WaitlistTableSchema.from_waitlist(wl) for wl in contact_waitlists
+    ]
```

Comment:
I'm kind of putting my entire initial review here because I think this section represents the "main idea" of this PR -- when we build a contact, include `created` and `updated` timestamps with waitlists and newsletters. Here are my thoughts:

- Is there a reason we only want to do this waitlist and newsletter "patching" for contacts here in this function? It seems like we'd want to include these everywhere, so we may want to change the `ContactSchema` from

```python
class ContactSchema(ComparableBase):
    """A complete contact."""
    ...
    newsletters: List[NewsletterSchema] = []
    waitlists: List[WaitlistSchema] = []
```

to 

```python
class ContactSchema(ComparableBase):
    """A complete contact."""
    ...
    newsletters: List[NewsletterTableSchema] = []
    waitlists: List[WaitlistTableSchema] = []
```

or add the missing fields directly to the existing schemas. 

In fact, saying that this function returns a `ContactSchema` might be "cheating" a bit since we define a Contact's newsletters and waitlists as `NewsletterSchema`s and `WaitlistSchema`s respectively, but now we're returning different models.

- if that doesn't work and we only want these new schemas here, I think that if they are truly a 1:1 mapping to the SQLAlchemy models, we can use Pydantic's [ORM Mode](https://docs.pydantic.dev/latest/usage/models/#orm-mode-aka-arbitrary-class-instances) rather than creating these custom class methods to convert ORM objects to Pydantic models